experimental/dds/tree2/src/feature-libraries/modular-schema/fieldChangeHandler.ts

```diff
@@ -156,7 +161,11 @@ export interface FieldEditor<TChangeset> {
  * The `index` should be `undefined` iff the child node does not exist in the input context (e.g., an inserted node).
  * @alpha
  */
-export type ToDelta = (child: NodeChangeset) => Delta.Modify;
+export type ToDelta = (
+	child: NodeChangeset,
+	repairDataBuilder: RepairDataBuilder,
+	idAllocator: IdAllocator,
```

Comment:
Could these parameters be automatically provided by the ModularChangeFamily implementation? This would save FieldKind implementation from having to pass them around.

Reply:
Actually it looks like your ModularChangeFamily changes already do what I had in mind, so I think these can just be removed because they're never used.

Reply:
They're meant to be used by the child field kind implementations. See the deltaFromChilds in `optionalFieldKind.spec.ts`. I did it this way cause I was thinking that any field kind implementation could need to generate repair deltas so we'd need to pass this info when generating child deltas. Is this not necessary or is there a way for the ToDeltas to access this from the ModularChangeFamily?

Reply:
You're indeed correct that any field kind implementation might need access to those, but this API is not about passing those to a field kind implementation, it is about passing those back to ModularChangeFamily (which provided them in the first place). If you look at the code you currently have in modularChangeFamily.ts, `intoDeltaImpl` doesn't use those parameters:
```
(childChange /* parms 2 and 3 are ignored */): Delta.Modify => this.deltaFromNodeChange(childChange, repairDataBuilder, idAllocator)
```

You are correct that removing those two parameters from `ToDelta` means that `deltaFromChild1` and `deltaFromChild2` need to change. I think you can make a helper function that take the RepairDataBuilder and the IdAllocator, and returns a `deltaFromChild` function that is seeded with the appropriate RepairDataBuilder and IdAllocator. It may even make sense for that helper function to build the RepairDataBuilder in the first place (and possibly the IdAllocator too, I'm not sure) and retrns them too.